libs/lwt_platform/dune

```diff
@@ -0,0 +1,26 @@
+;; What is this? And why are we doing it?
+;; Lwt uses unix threads to run its stuff on, which means it doesn't work in Js_of_ocaml
+;; This library uses Dune's virtual modules. The way this works is that code can use
+;; this library's interface, lwt_platform, and get type checking and everything without
+;; having to choose a specific implementation. Then, when you compile, we can choose
+;; which implementation to use.
+;;
+;; Why do we need this in general? There's a lot of [Lwt_main.run] calls in osemgrep
+;; in order to run async code. As mentioned, this uses unix threads, and so the underlying
+;; library, lwt.unix, pulls in unix dependencies. This means that we can't use lwt.unix
+;; in any code shared by osemgrep and LSP.js/turbo mode, etc., or else we'll get
+;; unix dependencies in the browser that won't work. So we can't make this a functor,
+;; as the shared networking code would have to be functorized over the Lwt implementation.
+;; Which means we'd have to pull in the lwt.unix dependency in the shared code, or use functors
+;; in like 5 or 6 different modules which would be a pain. We also can't use a ref for a similar reason,
```

Comment:
actually for the ref I think the issue is more than you can't have polymorphic refs at the toplevel

Reply:
That too